File: backend/tiggle-root/tiggle/src/main/java/com/side/tiggle/domain/category/dto/resp/CategoryRespDto.java
package com.side.tiggle.domain.category.dto.resp;

import com.side.tiggle.domain.asset.dto.resp.AssetRespDto;
import com.side.tiggle.domain.category.dto.CategoryDto;
import com.side.tiggle.domain.category.model.Category;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;

@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@SuperBuilder
public class CategoryRespDto extends CategoryDto {

    private Long id;

    public static CategoryRespDto fromEntity(Category category) {
        return CategoryRespDto.builder()
                .id(category.getId())
                .name(category.getName())
                .defaults(category.isDefaults())
                .build();
    }
}

File: backend/tiggle-root/tiggle/src/main/kotlin/com/side/tiggle/domain/category/dto/resp/CategoryRespDto.kt
package com.side.tiggle.domain.category.dto.resp

import com.side.tiggle.domain.category.model.Category

data class CategoryRespDto(
    val id: Long?,
    val name: String,
    val defaults: Boolean
) {

    companion object {
        fun fromEntity(category: Category?): CategoryRespDto {
            return when {
                category == null -> CategoryRespDto(
                    id = null,
                    name = "카테고리 없음",
                    defaults = false
                )

                else -> CategoryRespDto(
                    id = category.id,
                    name = category.name,
                    defaults = category.defaults
                )
            }
        }
    }
}
